--- hal-info/fdi/information/10freedesktop/10-usb-music-players.fdi.orig	2007-05-17 02:04:37.000000000 +0200
+++ hal-info/fdi/information/10freedesktop/10-usb-music-players.fdi	2007-05-24 23:18:34.699898742 +0200
@@ -215,6 +215,13 @@
             <append key="portable_audio_player.input_formats" type="strlist">audio/x-wav</append>
           </match>  
           <!-- Samsung YP-Z5 -->
+          <match key="@storage.physical_device:usb.product_id" int="0x503c">
+            <merge key="portable_audio_player.type" type="string">generic</merge>
+            <merge key="portable_audio_player.access_method" type="string">storage</merge>
+            <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+            <append key="portable_audio_player.output_formats" type="strlist">application/ogg</append>
+          </match>
+          <!-- Samsung YP-Z5 -->
           <match key="@storage.originating_device:usb.product_id" int="0x5041">
             <merge key="portable_audio_player.type" type="string">generic</merge>
             <merge key="portable_audio_player.access_method" type="string">storage</merge>
